### PR TITLE
geometry: Fix for rotation of Circle

### PR DESCRIPTION
**Pull request by rohitrango**
`rotate` function fixed for Circles
Example:

```
In [1]: cir = Circle(Point(1,0),1)

In [2]: cir.rotate(pi/2)
Out[2]: Circle(Point2D(0, 1), 1)

In [3]: cir.rotate(pi/3)
Out[3]: Circle(Point2D(1/2, sqrt(3)/2), 1)

In [4]: cir.rotate(pi/3,Point(1,0))
Out[4]: Circle(Point2D(1, 0), 1)

In [5]: cir.rotate(pi/3,Point(0,1))
Out[5]: Circle(Point2D(1/2 + sqrt(3)/2, 1/2 + sqrt(3)/2), 1)
```

### CODE REVIEW

**Comment by rohitrango** [comment]
@jksuom Could you review this please? There was a minor bug in the `rotate` method for ellipse. And do I have to write tests for this as well?

**Comment by jksuom** [comment]
This looks good to me. But a test would be welcome. It could be added to `test_ellipse_geom()` among the other rotate tests after a comment line referring to the issue.

**Comment by rohitrango** [comment]
I think this is ready to merge. The build is failing here but all tests pass on my laptop.

**Comment by asmeurer** [comment]
I restarted the build.

**Comment by rohitrango** [comment]
Thanks! It's good now. Why does the build fail a lot of times though?

File: sympy/geometry/tests/test_ellipse.py [hunk 1]
```diff
@@ -269,6 +269,13 @@ def line_list_close(ll1, ll2, prec):
     assert e.rotate(pi, (1, 2)) == Ellipse(Point(2, 4), 2, 1)
     raises(NotImplementedError, lambda: e.rotate(pi/3))
 
+    # Circle rotation tests (Issue #11743)
+    # Link - https://github.com/sympy/sympy/issues/11743
+    cir = Circle(Point(1, 0), 1)
+    assert cir.rotate(pi/2) == Circle(Point(0, 1), 1)
+    assert cir.rotate(pi/3) == Circle(Point(1/2, sqrt(3)/2), 1)
+    assert cir.rotate(pi/3,Point(1, 0)) == Circle(Point(1, 0), 1)
```
Comment (jksuom) [inline sympy/geometry/tests/test_ellipse.py]: A space after comma is expected on these lines, too.

**Comment by jksuom** [comment]
This is good to go. Thanks.

**Comment by rohitrango** [comment]
@jksuom was just wondering, when can I have my name in the contributors' list? I'm a little inactive now since my exams are going on.

**Comment by jksuom** [comment]
It seems that the AUTHORS list in the development version is infrequently updated. However, I would expect that to happen in a few weeks now, before the next release in any case.

**Pull request merged**